src/cache.ts

```diff
@@ -0,0 +1,82 @@
+import crypto from "node:crypto";
+import fs from "node:fs/promises";
+import path from "node:path";
+import { mkdirSync } from "node:fs";
+
+export class Cache {
+  private dir: string;
+  private loading: Record<string, Promise<unknown>>;
+  private isDisabled: boolean;
+
+  constructor(dir: string, isDisabled = false) {
+    this.dir = dir;
+    this.loading = {};
+    this.isDisabled = isDisabled;
+
+    if (!this.isDisabled) {
+      mkdirSync(this.dir, { recursive: true });
+    }
+  }
+
+  private key(key: string) {
+    return crypto.createHash("sha256").update(key).digest("hex");
+  }
+
+  private filename(key: string) {
+    return path.join(this.dir, this.key(key));
+  }
+
+  async get<T>(key: string): Promise<T | undefined> {
+    if (this.isDisabled) {
+      return undefined;
+    }
+
+    const filename = this.filename(key);
+
+    try {
+      return JSON.parse((await fs.readFile(filename)).toString()) as T;
```

Comment:
If you pass the encoding parameter to `readFile` (`readFile(filename, 'utf8')`), the `.toString()` should be unnecessary.